goosebit/updater/manager.py

```diff
@@ -137,11 +138,16 @@ def __init__(self, dev_id: str):
         super().__init__(dev_id)
         self.poll_time = config.poll_time_updating
 
-    async def _get_software(self) -> Software:
-        return await Software.latest(await self.get_device())
+    async def _get_software(self) -> Software | None:
+        device = await self.get_device()
+        if device is None:
+            return None
+        return await Software.latest(device)
 
-    async def get_update(self) -> tuple[HandlingType, Software]:
+    async def get_update(self) -> tuple[HandlingType, Software] | None:
         software = await self._get_software()
+        if software is None:
+            return None
```

Comment:
I think this can cause issues, I would rather return `HandlingType.SKIP, None`, and have the type be `tuple[Software | None, HandlingType]`